server/sonar-server/src/test/java/org/sonar/server/setting/ThreadLocalSettingsTest.java

```diff
@@ -233,6 +255,89 @@ public void cache_db_calls_if_property_is_not_persisted() {
     underTest.unload();
   }
 
+  @Test
+  public void getProperties_return_empty_if_DB_error_on_first_call_ever_out_of_thread_cache() {
+    SettingLoader settingLoaderMock = mock(SettingLoader.class);
+    PersistenceException toBeThrown = new PersistenceException("Faking an error connecting to DB");
+    doThrow(toBeThrown).when(settingLoaderMock).loadAll(any(ImmutableMap.Builder.class));
+    underTest = new ThreadLocalSettings(new PropertyDefinitions(), new Properties(), settingLoaderMock);
+
+    assertThat(underTest.getProperties())
+      .isEmpty();
+  }
+
+  @Test
+  public void getProperties_returns_empty_if_DB_error_on_first_call_ever_in_thread_cache() {
+    SettingLoader settingLoaderMock = mock(SettingLoader.class);
+    PersistenceException toBeThrown = new PersistenceException("Faking an error connecting to DB");
+    doThrow(toBeThrown).when(settingLoaderMock).loadAll(any(ImmutableMap.Builder.class));
+    underTest = new ThreadLocalSettings(new PropertyDefinitions(), new Properties(), settingLoaderMock);
+    underTest.load();
+
+    assertThat(underTest.getProperties())
+      .isEmpty();
+  }
+
+  @Test
+  public void getProperties_return_properties_from_previous_thread_cache_if_DB_error_on_not_first_call() {
+    String key = randomAlphanumeric(3);
+    String value1 = randomAlphanumeric(4);
+    String value2 = randomAlphanumeric(5);
```

Comment:
What's the benefit for using random values ? Test can't be reproduced when it fails.